### ISSUE TITLE
NETDEV net0 link is down

### ISSUE THREAD

**Issue by claell** [ipxe/ipxe#1099]
When trying to boot an iPXE ISO image, we get the following output (already using debug options):

![IMG_20231214_133117](https://github.com/ipxe/ipxe/assets/6171135/269ad6d9-0c1e-4b54-ae51-bdd6453d0fd7)

https://ipxe.org/err/380861 says:

> check that the cable is plugged in, and that any link lights are on

This should be the case.

The other link (https://ipxe.org/err/3c6540) points towards a driver problem?

Thus, I am writing here, to hopefully get a quick fix.

Am happy to provide more information like specific device info, etc.

iPXE version is the latest from GitHub master.

**Comment by NiKiZe** [comment]
Get to an iPXE shell and run ifstat, post output.
Also helpful if you have pciid of your NIC(s)

**Comment by claell** [comment]
How can I get into such a shell?

**Comment by NiKiZe** [comment]
Build without embedded script and press ctrl-b, or use shell in embedded script.

**Comment by claell** [comment]
Yup, found out that my ISO is probably with embedded script, so I used the prebuilt version from https://ipxe.org/download.

Below the output.

![IMG_20231214_165320](https://github.com/ipxe/ipxe/assets/6171135/e0efa4ca-21f9-40e4-ac97-1dec9e65fe41)

For PCI ID, I didn't find a quick way to see that on Windows. Not sure whether one of the screenshots from device manager contain the relevant information?

![IMG_20231214_170846](https://github.com/ipxe/ipxe/assets/6171135/dea6e488-8591-4f89-a920-bc218e28492a)
![IMG_20231214_170958](https://github.com/ipxe/ipxe/assets/6171135/1317bcce-c442-499e-93e1-7369467c68ee)
![IMG_20231214_171003](https://github.com/ipxe/ipxe/assets/6171135/31630bb5-00ce-4b53-948c-4635d3bade1e)

Also note, that we use a separate NIC, not the onboard card.
![IMG_20231214_171132](https://github.com/ipxe/ipxe/assets/6171135/c28e9413-2d9a-4627-af0c-4ecd7ab4ee24)

**Comment by NiKiZe** [comment]
So your machine has one intel 219LM-7 which is found by iPXE native drivers, and one Marvell 1D6A 07B1 which isn't supported by native drivers. Your simplest workaround is to plug your cable in to the supported NIC ;)

For this nic to be supported development needs to funded... However consider boot with efi instead, and use SNP.efi build of iPXE to use any NIC provided network stack instead.

**Comment by claell** [comment]
Alright. We already thought that this might be a solution, at least as workaround. But then, we couldn't have 10 Gigabit Ethernet.

Are there already Linux drivers available for that NIC? (i.e., how much effort is needed to support it, or how much funding for development)?

We'll also look into EFI. Will that then use the UEFI provided network interface (not an expert on this)?

**Comment by NiKiZe** [comment]
Development of iPXE drivers is done from scratch, hardware manuals for the NIC (chip) is required to do the work. See iPXE.org for commercial support.

If the NIC has a ROM that provides an efi network stack, then that will be used in EFI mode if no native support is available.

**Comment by claell** [comment]
Thanks for your quick support! We'll look into it and then decide, which way is the best.

**Comment by claell** [comment]
Adding further information for reference (after contacting commercial support): #370 seems to be the relevant PR for this.

**Comment by claell** [comment]
And this issue might then be a duplicate of #294?

**Comment by NiKiZe** [comment]
Same card as #294, which now has native drivers, difference between that issue and this is pcbios here and efi there but still same.
I also asume that this is working with current iPXE thru native drivers, while any SNP/UNDI issues must be fixed by the manufacturer.
